tests/Integration/Schema/Directives/AllDirectiveTest.php

```diff
@@ -249,6 +253,126 @@ public function testSpecifyCustomBuilderForRelation(): void
         ]);
     }
 
+    public function testSpecifyCustomBuilderForScoutBuilder(): void
+    {
+        $this->setUpScoutEngine();
+
+        /** @var \Tests\Utils\Models\Post $postA */
+        $postA = factory(Post::class)->create([
+            'title' => 'great title',
+        ]);
+        /** @var \Tests\Utils\Models\Post $postB */
+        $postB = factory(Post::class)->create([
+            'title' => 'Really great title',
+        ]);
+        factory(Post::class)->create([
+            'title' => 'bad title',
+        ]);
+
+        $this->engine->shouldReceive('map')
+            ->andReturn(
+                new EloquentCollection([$postA, $postB])
+            )
+            ->once();
+
+        $this->schema = /** @lang GraphQL */ <<<GRAPHQL
+        type Post {
+            id: ID!
+        }
+
+        type Query {
+            posts: [Post!]! @all(builder: "{$this->qualifyTestResolver('builderForScoutBuilder')}")
+        }
+GRAPHQL;
+
+        $this->graphQL(/** @lang GraphQL */ '
+        {
+            posts {
+                id
+            }
+        }
+        ')->assertJson([
+            'data' => [
+                'posts' => [
+                    [
+                        'id' => '1',
+                    ],
+                    [
+                        'id' => '2',
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    public function testSpecifyCustomBuilderForScoutBuilderWithScoutDirective(): void
+    {
+        $this->setUpScoutEngine();
+
+        /** @var \Mockery\MockInterface&\Laravel\Scout\Builder $builder */
+        $builder = $this->partialMock(\Laravel\Scout\Builder::class);
```

Comment:
Not supported in older PHPUnit/Laravel versions. Probably can be replaced by a check similar to `SearchDirectiveTest::testSearchWithEq()`.